File: Assignment1/Country.cpp
//David Tyler Gosbee
//201905838
//x2019fep@stfx.ca


#include "Country.h"


	Country::Country()
	{
		//Default constructor sets up Country object with default parameters
		_name = "DEFAULT";
		_pop = 0;
		_area = 0;
		_continent = "DEFAULT";
	}

	Country::Country(std::string name, int pop, double area, std::string continent)
	{
		//Constructor sets up Country object with parameters input
		_name = name;
		_pop = pop;
		_area = area;
		_continent = continent;
	}

	Country::~Country()
	{
		// No need for anything
	}

	std::string Country::getName()
	{
		//returns the name of the Country
		return _name;
	}

	int Country::getPopulation()
	{
		//returns the population of the Country
		return _pop;
	}

	double Country::getArea()
	{
		//returns the area of the Country
		return _area;
	}

	std::string Country::getContinent()
	{
		//returns the continent of the Country
		return _continent;
	}

	double Country::getPopulationDensity()
	{
		//returns the population density of the Country
		return _pop / _area;
	}

	void Country::setPopulation(int newPop)
	{
		//sets the population of the Country
		_pop = newPop;
	}

	std::string Country::to_string()
	{
		//returns a string form of the Country object
		return _name + " in " + _continent;
	}

File: Country.py

##
# Class: Country class. Each country has a name, population, area, and a continent it belongs to.
#
class Country:
    ##
    # Constructor
    #@param - name (name of Country)
    #@param - pop (population of Country)
    #@param - area (area of Country)
    #@param - continent (continent of Country)
    def __init__(self, name, pop, area, continent) :
        self._name=name #name of a country
        self._pop=pop #population of a country
        self._area=area #area of country
        self._continent=continent #continent of country
    ##
    # Convert to string a Country
    #
    def __repr__(self):
        return self._name + " in " + self._continent

    ##
    # sets the population of a country
    #@param - pop (a new population)
    #
    def setPopulation(self, pop):
        self._pop = pop
    ##
    # gets the name of the country
    #
    def getName(self):
        return self._name
    ##
    # gets the area of the country
    #
    def getArea(self):
        return self._area
    ##
    # gets the population of a country
    #
    def getPopulation(self):
        return self._pop
    ##
    # gets the continent of a country
    #
    def getContinent(self):
        return self._continent
    ##
    # computes the population density (pop/area) and returns this value.
    #
    def getPopDensity(self):
        return self._pop/self._area

    ##
    # given another country,
    #